src/ixdat/techniques/spectroelectrochemistry.py

```diff
@@ -0,0 +1,257 @@
+from .ec import ECMeasurement
+from ..spectra import Spectrum
+from ..data_series import Field, ValueSeries
+import numpy as np
+from scipy.interpolate import interp1d
+from ..spectra import SpectrumSeries
+from ..exporters.sec_exporter import SECExporter
+
+
+class SpectroECMeasurement(ECMeasurement):
+    def __init__(self, *args, **kwargs):
+        """Initialize an SEC measurement. All args and kwargs go to ECMeasurement."""
+        ECMeasurement.__init__(self, *args, **kwargs)
+        self._reference_spectrum = None
+        self.tracked_wavelengths = []
+        self.plot_waterfall = self.plotter.plot_waterfall
+        self.plot_wavelengths = self.plotter.plot_wavelengths
+        self.plot_wavelengths_vs_potential = self.plotter.plot_wavelengths_vs_potential
+        self.technique = "S-EC"
+
+    @property
+    def reference_spectrum(self):
+        """The spectrum which will by default be used to calculate dOD"""
+        if not self._reference_spectrum or self._reference_spectrum == "reference":
+            self._reference_spectrum = Spectrum.from_field(self["reference"])
+        return self._reference_spectrum
+
+    def set_reference_spectrum(
+        self,
+        spectrum=None,
+        t_ref=None,
+        V_ref=None,
+    ):
+        """Set the spectrum used as the reference when calculating dOD.
+
+        Args:
+            spectrum (Spectrum or str): If a Spectrum is given, it becomes the reference
+                spectrum. The string "reference" can be given to make the reference
+                spectrum become (via the reference_spectrum property) one that the
+                measurement was loaded with (evt. for definition of wavelengths).
+            t_ref (float): The time (with respect to self.tstamp) to use as the
+                reference spectrum
+            V_ref (float): The potential to use as the reference spectrum. This will
+                only work if the potential is monotonically increasing.
+        """
+        if (not spectrum) and t_ref:
+            spectrum = self.get_spectrum(t=t_ref)
+        if (not spectrum) and V_ref:
+            spectrum = self.get_spectrum(V=V_ref)
+        if not spectrum:
+            raise ValueError("must provide a spectrum, t_ref, or V_ref!")
+        self._reference_spectrum = spectrum
+
+    @property
+    def spectra(self):
+        """The Field that is the spectra of the SEC Measurement"""
+        return self["spectra"]
+
+    @property
+    def spectrum_series(self):
+        """The SpectrumSeries that is the spectra of the SEC Measurement"""
+        return SpectrumSeries.from_field(
+            self.spectra,
+            tstamp=self.tstamp,
+            name=self.name + " spectra",
+        )
+
+    @property
+    def wavelength(self):
```

Comment:
what is the difference for wavelength and wl, why need one dataseries and one np array?